src/screens/SendSheet.js

```diff
@@ -475,16 +492,18 @@ export default function SendSheet(props) {
         ? updatedGasLimit
         : gasLimit;
 
-    logger.log('gasLimit', gasLimitToUse);
+    const gasParams = parseGasParamsForTransaction(selectedGasFee);
     const txDetails = {
       amount: amountDetails.assetAmount,
       asset: selected,
       from: accountAddress,
       gasLimit: gasLimitToUse,
-      gasPrice: selectedGasPrice.value?.amount,
+      network: assetNetwork,
```

Comment:
why can't `currentNetwork` be used here?

Reply:
is my understanding correct that in line 331 bc of `selected?.type &&` that the currentNetwork would not ever be set to `mainnet`? (assuming `type` doesn't exist on `selected` for `mainnet`?)

Reply:
imo `setCurrentNetwork` should be set anytime the `selected` changes (even if `selected` is on `mainnet`)

Reply:
line 136: is there any harm in using a default of `useState(network)` in `const [currentNetwork, setCurrentNetwork] = useState();` ?

Reply:
- `why can't currentNetwork be used here?` change it to use it
- `is my understanding correct that in line 331 ...` yeah seems like that, but the type for mainnet assets is `token` or `nft` so it should never be undefined
- `imo setCurrentNetwork should be set anytime the selected...` i believe we only don't do that to not initialize another provider, only triggering that when a network is changed 
- `line 136: is there any harm in using a default of useState(network)...` nope, adding it

Reply:
`line 136: is there any harm in using a default of useState(network)...` ah yes, we need to set provider once the user gets to this screen, if `prevNetwork === currentNetwork` the gas limit estimation won't run

Reply:
@estebanmino let's catch up on this